src/test/scala/io/iohk/ethereum/faucet/FaucetHandlerSpec.scala

```diff
@@ -0,0 +1,157 @@
+package io.iohk.ethereum.faucet
+
+import java.security.SecureRandom
+
+import akka.actor.{ActorSystem, Props}
+import akka.pattern.gracefulStop
+import akka.testkit.{ImplicitSender, TestKit, TestProbe}
+import akka.util.ByteString
+import io.iohk.ethereum.crypto.{generateKeyPair, keyPairToByteStrings}
+import io.iohk.ethereum.domain.Address
+import io.iohk.ethereum.faucet.FaucetHandler.{FaucetHandlerMsg, FaucetHandlerResponse}
+import io.iohk.ethereum.faucet.jsonrpc.WalletService
+import io.iohk.ethereum.keystore.KeyStore.DecryptionFailed
+import io.iohk.ethereum.keystore.Wallet
+import io.iohk.ethereum.mallet.common.{ParserError, RpcClientError}
+import io.iohk.ethereum.{NormalPatience, WithActorSystemShutDown, crypto}
+import monix.eval.Task
+import org.bouncycastle.util.encoders.Hex
+import org.scalamock.scalatest.MockFactory
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.freespec.AnyFreeSpecLike
+import org.scalatest.matchers.should.Matchers
+
+import scala.concurrent.ExecutionContext
+
+class FaucetHandlerSpec
+    extends TestKit(ActorSystem("ActorSystem_DebugFaucetHandlerSpec"))
+    with AnyFreeSpecLike
+    with ImplicitSender
+    with WithActorSystemShutDown
+    with Matchers
+    with MockFactory
+    with ScalaFutures
+    with NormalPatience {
+
+  "Faucet Handler" - {
+    "without wallet unlocked" - {
+
+      "should try to unlock the Wallet if it is not initialized but decryption failed" in new TestSetup {
```

Comment:
For me this test should like:
"Faucet Handler" + "without wallet unlocked"  + "should not respond in case wallet unlock fails"

Reply:
Done